app/backend/src/couchers/models.py

```diff
@@ -239,6 +243,60 @@ def __repr__(self):
         return f"User(id={self.id}, email={self.email}, username={self.username})"
 
 
+# CouchersQuery is placed here (instead of the more natural db) because we import stuff from models into db, so need to
+# avoid circular imports. Additionally, we need it in the relationship configuration, but it requires the User model, so
+# it's as high up as we can get it, and then we patch the User relationships in manually.
```

Comment:
I really think `CouchersQuery` makes everything *a lot* nicer, cleaner, and more concise than having separate functions that you have to wrap around your query. In some of the bigger queries, they just become a huge mess otherwise, especially if you have to use multiple filters.

> Or introduce a third file to avoid circular imports.

The problem is that:
* `CouchersQuery` requires the `User` class to be defined
* `relationship` requires `CouchersQuery` to be defined
* Basically all classes require `relationship` to be defined

I'm not sure how to get around this, but it'd be really nice to find a cleaner way.

Maybe if we split `User` into another file? The split of `models.py` is long overdue, but those PRs never seem to go through and somehow become stale so the merges are a pain...